src/config/mod.rs

```diff
@@ -1196,21 +1240,87 @@ pub enum Error {
     Yaml(serde_yaml::Error),
 }
 
-#[derive(Debug, Deserialize)]
+#[serde(default)]
+#[derive(Debug, Deserialize, PartialEq, Eq)]
 pub struct Colors {
-    #[serde(default, deserialize_with = "failure_default")]
+    #[serde(deserialize_with = "failure_default")]
     pub primary: PrimaryColors,
-    #[serde(default, deserialize_with = "failure_default")]
+    #[serde(deserialize_with = "failure_default")]
     pub cursor: CursorColors,
+    #[serde(deserialize_with = "deserialize_normal_colors")]
     pub normal: AnsiColors,
+    #[serde(deserialize_with = "deserialize_bright_colors")]
     pub bright: AnsiColors,
-    #[serde(default, deserialize_with = "failure_default")]
+    #[serde(deserialize_with = "failure_default")]
     pub dim: Option<AnsiColors>,
-    #[serde(default, deserialize_with = "failure_default_vec")]
+    #[serde(deserialize_with = "failure_default_vec")]
     pub indexed_colors: Vec<IndexedColor>,
 }
 
-#[derive(Debug, Deserialize)]
+impl Default for Colors {
+    fn default() -> Colors {
+        Colors {
+            primary: Default::default(),
+            cursor: Default::default(),
+            normal: default_normal_colors(),
+            bright: default_bright_colors(),
+            dim: Default::default(),
+            indexed_colors: Default::default(),
+        }
+    }
+}
+
+fn default_normal_colors() -> AnsiColors {
+    AnsiColors {
+        black: Rgb {r: 0x00, g: 0x00, b: 0x00},
+        red: Rgb {r: 0xd5, g: 0x4e, b: 0x53},
+        green: Rgb {r: 0xb9, g: 0xca, b: 0x4a},
+        yellow: Rgb {r: 0xe6, g: 0xc5, b: 0x47},
+        blue: Rgb {r: 0x7a, g: 0xa6, b: 0xda},
+        magenta: Rgb {r: 0xc3, g: 0x97, b: 0xd8},
+        cyan: Rgb {r: 0x70, g: 0xc0, b: 0xba},
+        white: Rgb {r: 0xea, g: 0xea, b: 0xea},
+    }
+}
+
+fn default_bright_colors() -> AnsiColors {
+    AnsiColors {
+        black: Rgb {r: 0x66, g: 0x66, b: 0x66},
+        red: Rgb {r: 0xff, g: 0x33, b: 0x34},
+        green: Rgb {r: 0x9e, g: 0xc4, b: 0x00},
+        yellow: Rgb {r: 0xe7, g: 0xc5, b: 0x47},
+        blue: Rgb {r: 0x7a, g: 0xa6, b: 0xda},
+        magenta: Rgb {r: 0xb7, g: 0x7e, b: 0xe0},
+        cyan: Rgb {r: 0x54, g: 0xce, b: 0xd6},
+        white: Rgb {r: 0xff, g: 0xff, b: 0xff},
+    }
+}
```

Comment:
Here's a good argument against using a `Default` trait of any kind, since as you've found there is no sensible default for `AnsiColors` because we use it in two places within the config. It looks a little weird to have some things with `impl Default` and others with `fn default_`, but I'm being really picky (as usual).

I suppose the `ConfigDefault` trait could have a function like `config_default(key: String) -> Self` but IDK.

Just more food for thought.

Reply:
I don't see how this is an argument to not use the Default trait at all? For AnsiColors it doesn't make sense, or it's not useful to us, to have the Default trait implemented, so it is not. If the Default implementation makes sense, it should be implemented (or derived).

The reason why some things have `impl Default` and others have `fn default_` is simply because it makes sense for some **types** to implement `Default`, while other times it is not and it only makes sense to have defaults for **fields**.